### PR TITLE
Additional bash completions

### PR DESCRIPTION
**Pull request by ghost**
Added completions for new `docker service scale [SERVICE NAME/ID]=` short cut command.

### CODE REVIEW

File: contrib/completion/bash/docker [hunk 1]
```diff
@@ -1597,8 +1598,18 @@ _docker_service_rm() {
...
+	esac
+}
+
+_docker_service_scale() {
+	case "$cur" in
+		-*)
 			COMPREPLY=( $( compgen -W "--help" -- "$cur" ) )
 			;;
+		*)
+			COMPREPLY=( $(compgen -S "=" -o nospace -W "$(__docker_services $1)" -- "$cur") )
```
Comment (mgoelzer) [inline contrib/completion/bash/docker]: @albers Do you know why `-o nospace` is not being obeyed?  

This PR will complete `docker service scale m<TAB>` into `docker service scale myservice= `.  But the trailing equals is unacceptable.

**Comment by vieux** [comment]
do we know how it works ? those commands are experimental, won't it cause issues when you have the stable build ?

**Comment by mgoelzer** [comment]
`docker service scale` is not experimental.  It's the only thing added in this PR.

@albers I haven't found a fix for the `nospace` issue (see line 1611 comment above).  It's annoying to have to type <TAB> then backspace to fill in the instance count, but I don't know of a good solution.

File: contrib/completion/bash/docker (hunk 1 above)
Comment (albers) [inline contrib/completion/bash/docker]: @mgoelzer you have to use `compopt -o nospace` as a separate command. But this does not work on Macs, so the idiom used here is `__docker_nospace`, see https://github.com/docker/docker/blob/master/contrib/completion/bash/docker#L980 for an example.

File: contrib/completion/bash/docker [hunk 2]
```diff
@@ -1597,8 +1598,18 @@ _docker_service_rm() {
 _docker_service_ls() {
 	case "$cur" in
 		-*)
+			COMPREPLY=( $( compgen -W "--help --filter --quiet" -- "$cur" ) )
```
Comment (albers) [inline contrib/completion/bash/docker]: please keep the options in alphabetical order

**Comment by mgoelzer** [comment]
Fixed alphabetical order and `__docker_no_space`

@albers Thank you for your help!

**Comment by thaJeztah** [comment]
LGTM

**Comment by albers** [comment]
Please do not include unrelated changes in this PR.
I am currently working on a follow-up PR that will also update the options of the other commands.
This will cause lots of merge conflicts.

**Comment by mgoelzer** [comment]
@albers 

I didn't add any unrelated changes in this PR (except alphabetization, which you requested).

If you like, I can close this PR and yours can make all the fixes?

**Comment by albers** [comment]
@mgoelzer Oh, sorry, I was not clear enough. I was thinking about the alphabetization in just the added piece.

> If you like, I can close this PR and yours can make all the fixes?

OK, as you will be on the road for one week (https://github.com/docker/docker/pull/23665#issuecomment-226717992), I'll carry this one.

**Comment by mgoelzer** [comment]
@albers Either is fine with me.  I just didn't want to commit to getting it finished quickly and then fail to deliver.  I think it's better for you to handle it.  Feel free to close or carry it.  Thank you.

**Comment by thaJeztah** [comment]
Let me close this one then, thanks @albers!

**Pull request closed without merging**